src/js/components/agencyV2/statusOfFunds/StatusOfFunds.jsx

```diff
@@ -87,19 +133,20 @@ const StatusOfFunds = ({ fy }) => {
         }
     }, [fy, overview.toptierCode]);
 
-    // TODO - remove mock data when DEV-8052 is implemented
-    const mockData = {
-        name: "Bureau of the Census",
-        id: "bureau_of_the_census",
-        total_budgetary_resources: 5000000,
-        total_obligations: 3000000.72
+    const onClick = (selectedLevel, data) => {
+        // reset to page 1 on drilldown
+        changeCurrentPage(1);
```

Comment:
This seems to be causing the issue with the delayed update mentioned.   New data is fetched and the chart is updated here in response to the page change, but also get's updated due to the subcomponent change that happens later in this function.